### PR TITLE
Jwt

### PR DESCRIPTION
**Pull request by oleksii-sukhenko**

### CODE REVIEW

File: src/main/java/mate/academy/dto/user/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package mate.academy.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/user/UserLoginRequestDto.java]:
```suggestion
        @NotBlank
```

File: src/main/java/mate/academy/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        @Email
+        String email,
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/user/UserLoginRequestDto.java]:
```suggestion
        @NotBlank
```

File: src/main/java/mate/academy/model/User.java [hunk 3]
```diff
@@ -34,7 +35,7 @@ public class User implements UserDetails {
     @Column(nullable = false)
     private String lastName;
     private String shippingAddress;
-    @ManyToMany
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/User.java]:
```suggestion
    @ManyToMany
```

File: src/main/java/mate/academy/security/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/security/AuthenticationService.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/model/User.java (hunk 3 above)
Comment (oleksii-sukhenko) [reply src/main/java/mate/academy/model/User.java]: I have LazyInitializeException without fetchType.EAGER...

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**